src/irmin-pack/unix/dispatcher_intf.ml

```diff
@@ -50,6 +50,38 @@ module type S = sig
   (** [shrink_accessor_exn a ~new_len] is [a] where the length is smaller than
       in [a].*)
 
+  module Sequential : sig
+    val create_accessor_exn : int63 -> int63 -> off:int63 -> len:int -> accessor
```

Comment:
The prefix & suffix length are functions that are very costly (or at least for the prefix as it is a syscall), therefore I took the choice to give those directly to the accessors creators.

Maybe a cleaner solution would be to save those in the `Dispatcher.t`, however it does mean that we would calculate them (and have to handle the potential errors) even when the user will never use the accessors that needs them.

Dont hesitate if you have a better solution

Reply:
It makes sense to avoid recomputing the prefix length over and over. 

However, is there any real use for exposing `create_accessor_exn` and `create_accessor_from_range_exn` in the `Dispatcher` signature? Since they introduce a third kind of offset it could be great if that concept stayed hidden inside the implementation of `Sequential` (you could preserve your docstrings to the `.ml` files).

Reply:
At first I had them hidden, but for the purpose of [#1939](https://github.com/mirage/irmin/pull/1939), I'll also need direct access using the offset returned